### ISSUE TITLE
Support `*.mol` file

### ISSUE THREAD

**Issue by Acylation** [Acylation/obsidian-chem#69]
### What would you like to be added?

Powered by the awesome RDKit.js, we can process more file formats. This request aims to load `*.mol` file from vault and display them as embed/still codeblock.

This feature requires a better structure
- Separated parser and renderer
- Obsidian embed syntax
- Architecture that allow choosing from multiple parsers

### Why is this needed?

This would be useful for computational/theoretical guys.

**Comment by RyougiShiki** [comment]
.mol structure format is much useful than SMILES since the coordinate of every atoms in the structure is defined, and you can draw reaction arrows in this format.

**Comment by realhackcraft** [comment]
I agree that it is more useful. If you want reaction arrows in SMILES, try the format `Reactants > Agents/Catalysts > Products`.

Note that you can't specify catalysts with the `RDKit` renderer.

For example:
`C=C.Cl>O>CCCl`

With SmilesDrawer is 
![image](https://github.com/user-attachments/assets/b34da5d7-32bb-4182-bdd3-5a3bcebf605c)
